src/main.rs

```diff
@@ -14,13 +15,46 @@ fn main() -> Result<(), Box<dyn Error>> {
         process::exit(0);
     }
 
-    let warnings = dotenv_linter::run(args, &current_dir)?;
+    let warnings = dotenv_linter::run(&args, &current_dir)?;
     if warnings.is_empty() {
         process::exit(0);
     }
 
-    warnings.iter().for_each(|w| println!("{}", w));
-    process::exit(1);
+    print_warnings(args.is_present("no-color"), warnings);
+    process::exit(1)
+}
+
+fn print_warnings(no_color: bool, warnings: Vec<dotenv_linter::common::Warning>) {
+    if no_color {
```

Comment:
It seems to be better to create two separate functions for _color output_ and _no-color_. 
This will make the code much easier to read.